packages/venia-ui/lib/components/CategorySort/categorySort.js

```diff
@@ -0,0 +1,125 @@
+import React, { useMemo, useState } from 'react';
+import { Check } from 'react-feather';
+import uuidV4 from 'uuid/v4';
+import { arrayOf, shape, string, func, oneOf } from 'prop-types';
+
+import defaultClasses from './categorySort.css';
+import { mergeClasses } from '../../classify';
+import Icon from '../Icon/icon';
+
+const CategorySort = props => {
+    const classes = mergeClasses(defaultClasses);
+    const { availableSortMethods, sortControl } = props;
+    const { currentSort, setSort } = sortControl;
+
+    const [visible, setVisible] = useState(false);
+    const [sortAttribute, setSortAttribute] = useState(
+        currentSort.sortAttribute
+    );
+    const [sortDirection, setSortDirection] = useState(
+        currentSort.sortDirection
+    );
+
+    const sortElements = useMemo(() => {
+        const onChildButtonClick = sortAttribute => {
+            setSort({
+                sortAttribute: sortAttribute.attribute,
+                sortDirection: sortAttribute.sortDirection
+            });
+            setVisible(false);
+            setSortAttribute(sortAttribute.attribute);
+            setSortDirection(sortAttribute.sortDirection);
+        };
+
+        if (visible === false) {
+            return;
+        }
+
+        const sortElements = [];
+        for (const element of availableSortMethods) {
+            const key = uuidV4();
+            sortElements.push(
+                <li key={key} className={classes.child}>
+                    <button
+                        className={classes.childButton}
+                        onClick={onChildButtonClick.bind(null, element)}
```

Comment:
If you search the codebase for `.bind(`, you won't find any results. 😅 

In years past, binding handlers was fairly common practice in React, but it's no longer recommended. I recommend creating a component specifically for these buttons, and letting that component define its own `handleClick` handler (and memoize it with `useCallback`!). There should be a number of examples in the codebase you can use for reference.